eth/tracers/api_blocktrace.go

```diff
@@ -109,5 +110,21 @@ func (api *API) createTraceEnvAndGetBlockTrace(ctx context.Context, config *Trac
 	}
 
 	chaindb := api.backend.ChainDb()
-	return api.scrollTracerWrapper.CreateTraceEnvAndGetBlockTrace(api.backend.ChainConfig(), api.chainContext(ctx), api.backend.Engine(), chaindb, statedb, parent, block, true)
+	// create a copy of api.backend.ChainConfig to modify
+	chainConfig := new(params.ChainConfig)
+	*chainConfig = *api.backend.ChainConfig()
+	if config != nil && config.Overrides != nil {
+		// In future we can add more fork related logics here
+		// like upstream: https://github.com/ethereum/go-ethereum/pull/26655
+		if curie := config.Overrides.CurieBlock; curie != nil {
+			chainConfig.CurieBlock = curie
+			if block.Number().Cmp(curie) > 0 {
+				// set non zero values for these slots
+				misc.ApplyCurieHardFork(statedb)
```

Comment:
if Curie fork is already enabled, this will overwrite `L1GasPriceOracle`'s storage.

Reply:
`if block.Number().Cmp(api.backend.ChainConfig().CurieBlock) < 0 && block.Number().Cmp(curie) > 0 {`. sounds more reasonable?

Reply:
Yeah, that is better. But I am afraid not complete. Maybe something like below?

> !api.backend.ChainConfig().IsCurie(block.Number()) && block.Number().Cmp(curie) >= 0`

Reply:
in fact if block.Number().Cmp(curie) == 0, i think we don't need to inject system upgrades twice.

Reply:
> in fact if block.Number().Cmp(curie) == 0, i think we don't need to inject system upgrades twice.

as long as `!api.backend.ChainConfig().IsCurie(block.Number())` is true, I don't think we would be applying the upgrade twice. If we apply only when `block.Number().Cmp(curie) > 0`, ApplyCurieHardFork will be applied one block late if I am not mistkaen.

Reply:
"ApplyCurieHardFork will be applied one block late": you mean the bytecode force upgrade does not happen exactly at the fork height?  i don't think so..